src/pgp.ts

```diff
@@ -0,0 +1,92 @@
+import { createReadStream, promises as fs } from "fs";
+import * as openpgp from "openpgp";
+import * as path from "path";
+import { Readable } from "stream";
+import * as vscode from "vscode";
+import { errToStr } from "./api-helper";
+
+export type Key = openpgp.Key;
+
+export enum VerificationErrorCode {
+	/* The signature does not match. */
+	Invalid = "Invalid",
+	/* Failed to read the signature or the file to verify. */
+	Read = "Read",
+}
+
+export class VerificationError extends Error {
+	constructor(
+		public readonly code: VerificationErrorCode,
+		message: string,
+	) {
+		super(message);
+	}
+
+	summary(): string {
+		switch (this.code) {
+			case VerificationErrorCode.Invalid:
+				return "Signature does not match";
+			default:
+				return "Failed to read signature";
```

Comment:
```suggestion
				return "Failed to verify signature";
```

since this is the default, it might not necessarily be about reading I guess?

Reply:
Ah true, there are only two error codes and the other is `read` so as it currently is this is accurate, but if in the future a code was added and someone did not update the switch statement, we could possibly return the wrong message.  I think I can make use of `never` instead in some way to make this impossible, will fix

Reply:
Waaaaiiit looks like TypeScript is smart about this now??  No need to do anything special.  When did that happen lol

So I can just replace `default` with `VerificationErrorCode.Read` and all is well.